src/cnxml/jsx.ts

```diff
@@ -185,6 +186,11 @@ function finishElement(renderer: Renderer, element: Element, out: globalThis.Ele
             out.append(renderer.doc.importNode(child, true))
         } else if (typeof child === 'string') {
             out.append(child)
+        } else if (child.name.local === 'pi') {
+            out.append(document.createProcessingInstruction(
+                child.attributes.target as string,
+                child.attributes.value as string,
+            ))
```

Comment:
This would prevent us from ever using an element named `pi`. It also confuses processing instructions with elements. Support for processing instructions should be more explicit:

1. Add new type for processing instructions:

   ```ts
   /** A processing instruction */
   export interface ProcessingInstruction {
      /** PI's target */
      target: string
      /** PI's value */
      value: string
   }
   ```

2. Add it as a possible `Node`:

   ```diff
    /** Any value that can be used as child of a JSX element */
   -export type Node = Element | globalThis.Node | string | Node[] | null
   +export type Node = Element | ProcessingInstruction | globalThis.Node | string | Node[] | null
   ```

3. Handle this new type in `renderChild`

   ```diff
    if (child instanceof globalThis.Node) {
       out.append(renderer.doc.importNode(child, true))
    } else if (typeof child === 'string') {
       out.append(child)
   +} else if (typeof child.target === 'string') {
   +   out.append(document.createProcessingInstruction(child.target, child.value))
    } else {
       out.append(renderElement(r, child))
    }
   ```